packages/conf-ida/conf-ida.0.2/files/find-ida.ml.in

```diff
@@ -56,81 +91,92 @@ let exn_to_string = function
   | Command_failed s -> sprintf "%s" (process_status_to_string s)
   | e -> Printexc.to_string e
 
-let cmd fmt =
-  let run c =
-    try
-      let inp = Unix.open_process_in c in
-      let res = read_all inp in
-      match Unix.close_process_in inp with
-      | Unix.WEXITED 0 -> Some res
-      | s -> raise (Command_failed s)
-    with e ->
-      eprintf "command %s failed: %s\n" c (exn_to_string e);
-      None in
-  ksprintf run fmt
-
-let find_map xs ~f =
-  let rec loop = function
-    | [] -> None
-    | hd :: tl ->
-      match f hd with
-      | None -> loop tl
-      | x -> x in
-  loop xs
-
-let known_names = ["ida"; "idaq"]
-
-let which_ida64 () = find_map known_names ~f:(cmd "which %s64")
-
-let find_ida64 path =
-  cmd
-    "find %s -executable -type f -name \"ida*64\" 2>/dev/null | sort -n -r"
-    path |> function
-  | None -> None
-  | Some idas ->
-    match split idas with
-    | [] -> None
-    | idas ->
-      let known = List.map (fun x -> x ^ "64") known_names in
-      find_map idas ~f:(fun file ->
-          if List.mem (Filename.basename file) known then
-            Some file
-          else None)
+let run c =
+  try
+    let inp = Unix.open_process_in c in
+    let res = read_all inp in
+    match Unix.close_process_in inp with
+    | Unix.WEXITED 0 -> Some res
+    | s -> raise (Command_failed s)
+  with e ->
+    eprintf "command %s failed: %s\n" c (exn_to_string e);
+    None
+
+let cmd fmt = ksprintf run fmt
+
+let known_names = ["ida"; "idaq";]
+let known_names64 = List.map known_names ~f:(fun x -> x ^ "64")
+
+let which_ida64 () = List.find_map known_names64 ~f:(cmd "which %s")
+let is_dir x = Sys.file_exists x && Sys.is_directory x
+
+let all_pathes () =
+  match getenv_opt "PATH" with
+  | None -> []
+  | Some r -> List.filter ~f:is_dir (split r ~on:':')
+
+let contains_ida64 path =
+  Sys.readdir path |>
+  Array.exists ~f:(fun file -> List.mem ~set:known_names64 file)
 
 let find_at_home () =
   match getenv_opt "HOME" with
   | None -> None
-  | Some home -> find_ida64 home
+  | Some home ->
+    Sys.readdir home |>
+    Array.find ~f:(fun p -> contains_ida64 @@ home / p)
+
+let is_some = function
+  | None -> false
+  | Some _ -> true
 
 let locate_linux () =
-  match find_at_home () with
-  | None -> find_ida64 "/"
-  | x -> x
+  match List.find_opt ~f:contains_ida64 @@ all_pathes () with
```

Comment:
You shouldn't be searching in the `PATH` constituents, you shall use `which` for that. Use linear search only for paths, that are not in `PATH`.